pandas/core/groupby.py

```diff
@@ -642,9 +642,20 @@ def apply(self, func, *args, **kwargs):
 
         func = self._is_builtin_func(func)
 
-        @wraps(func)
-        def f(g):
-            return func(g, *args, **kwargs)
+        # this is needed so we don't try and wrap strings. If we could
+        # resolve functions to their callable functions prior, this
+        # wouldn't be needed
+        if args or kwargs:
+            if is_callable(func):
+
+                @wraps(func)
+                def f(g):
+                    return func(g, *args, **kwargs)
+            else:
+                raise ValueError('func must be a callable if args or '
+                                 'kwargs are supplied')
+        else:
+            f = func
```

Comment:
This is messy. I think would be better to resolve to functions _before_ this (or throughout - not sure why we go to strings at all). But that's for discussion / another PR